coderd/autobuild/executor/lifecycle_executor_test.go

```diff
@@ -200,6 +204,10 @@ func TestExecutorAutostopOK(t *testing.T) {
 	assert.Len(t, stats.Transitions, 1)
 	assert.Contains(t, stats.Transitions, workspace.ID)
 	assert.Equal(t, database.WorkspaceTransitionStop, stats.Transitions[workspace.ID])
+
+	workspace, err := client.Workspace(context.Background(), workspace.ID)
+	assert.NoError(t, err)
+	assert.Equal(t, codersdk.BuildReasonAutostop, workspace.LatestBuild.Reason)
```

Comment:
You may need to wait for the build job to complete in order for this to work!
`coderdtest.AwaitWorkspaceBuildJob`

Reply:
Actually I don't think he does. The build is inserted before the stat is returned. We don't care if the build is done or not, just checking the reason for it.

Reply:
+1 to what Steven says.